nixos/modules/services/hardware/dell-bios-fan-control.nix

```
@@ -0,0 +1,48 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+let
+  cfg = config.services.hardware.dell-bios-fan-control;
+in
+{
+  meta.maintainers = with lib.maintainers; [ rickyelopez ];
+
+  options.services.hardware.dell-bios-fan-control = {
+    enable = lib.mkEnableOption "One-shot service to disable dell bios fan control on startup";
+    package = lib.mkPackageOption pkgs "dell-bios-fan-control" { };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = [ cfg.package ];
+    # see ref in aur: https://aur.archlinux.org/cgit/aur.git/tree/dell-bios-fan-control.service?h=dell-bios-fan-control-git
+    systemd.services.dell-bios-fan-control = {
+      description = "Disables BIOS control of fans at boot.";
+      wants = [ "dell-bios-fan-control-resume.service" ];
+      wantedBy = [ "multi-user.target" ];
+      before = [ "i8kmon.service" ];
```

I don't think we have this service in nixpkgs? I suppose the kernel module is still needed for this to work though. There is a shiny new condition for kernel modules in systemd v258 that we could use.

```suggestion
      unitConfig.ConditionKernelModuleLoaded = "i8k";
```